src/modules/party-external-rating/party-external-rating.controller.ts

```diff
@@ -0,0 +1,48 @@
+import { Controller, Get, Inject, Param } from '@nestjs/common';
+import { ApiInternalServerErrorResponse, ApiNotFoundResponse, ApiOkResponse, ApiOperation, ApiParam } from '@nestjs/swagger';
+
+import { GetPartyExternalRatingResponse, GetPartyExternalRatingsResponse } from './dto/get-party-external-ratings-response.dto';
+import { PARTY_EXTERNAL_RATINGS_PROVIDER_SYMBOL } from './party-external-rating.module-definition';
+import { PartyExternalRatingsProvider } from './party-external-ratings.provider';
+
+@Controller()
```

Comment:
Do we anticipate orgainzing all operations relating to a party resource into one controller? ie. the party is the 'root' resource of an external rating, the party external rating is a property of a party.

Just wondering from the name choice of this class, which seems to imply this will only handle operations relating to a party external rating. Would we continue with a "one controller per endpoint approach", or maybe do something like this? -- 
```
@Controller('party')
export class PartyController {
  ...
  @Get(':partyIdentifier')
  async foo() { ... }
  
  @Get(':partyIdentifier/external-rating)
  async bar() { ... }
```

Reply:
> Do we anticipate orgainzing all operations relating to a party resource into one controller? ie. the party is the 'root' resource of an external rating, the party external rating is a property of a party.

I don't think there's any particular need for having all operations relating to a party resource and any child resource in one controller. From past experience, I've found that this leads to overly large controllers and that there's no harm to splitting them if you name the controllers sensibly.

I'm expecting we'll end up with something like
- PartyController:
   - handles `GET /party?searchtext=`
   - handles `GET /party/{partyId}`
   - handles `POST /party`
- PartyExternalRatingController:
   - handles `GET /party/{partyIdentifier}/external-rating`
   - handles `POST /party/{partyIdentifier}/external-rating`
 
We can always put both of these into a folder called `party` if we feel like we need to emphasise the root entity (more than starting the controller name with `Party`)

Looking ahead, I'm expecting we'll have 22 endpoints where `facility` is the "root" resource, and this will be very unwieldy if we insist on putting all these endpoints in the same controller :)

Reply:
good point @catenoid  - that's definitely something we need to take into account.

In terms of approach here, I agree with Luke - most likely the file will start to grow by a lot. I'm happy to split them where we thing it's appropriate